gui_server_ecm_sync.md

```diff
@@ -0,0 +1,56 @@
+# Synchronization Between the GUI and Server ECM
+
+## Goal
+When creating, deleting, or modifying entities/components in the GUI _while paused_,
+GUI changes should be reflected in the server once simulation is resumed, or stepped while paused.
+
+## Approach
+If a user performs actions that modify/create/remove entities while simulation is paused, the ECM on the GUI side is updated directly to reflect these actions.
+The server's ECM does not reflect these user actions yet, but will be notified of the changes that took place once simulation is resumed or stepped while paused.
+When the server ECM is notified of these changes, it will be updated to have any changes made to the GUI ECM (unless the GUI and server are run in the same process, which will be discussed below).
+Once the server ECM is updated, simulation will be carried out as normal.
+
+The image below depicts how changes to the GUI's ECM are propogated to the server's ECM.
+The GUI and server interaction can be done through an event (green arrows), or through a service (red arrow).
+If GUI and server interaction is done through a service, then there's no way to propogate GUI ECM changes to the server.
+The black arrows in the image are things that occur in both the event and service approach.
+
+![gui_server_ecm_design](images/gui_server_ECM_sync_design.png)
```

Comment:
Some comments about the image:

* It would be nice to trace the GUI / server boundary
* It's not clear to me why the `WorldStatsTopic` gets a box, instead of being an arrow. 
* It would be nice to display the topic names

Reply:
I updated the image in https://github.com/ignitionrobotics/design/pull/8/commits/dce6cae9cba7926636d5252dc53d37e636d89e25, let me know what you think!